ps_deployment/README.md

```diff
@@ -0,0 +1,121 @@
+
+### Image build approach
+	* s2i:
+		Source-to-Image (S2I) is a tool for building reproducible, Docker-formatted container images. It produces ready-to-run images by injecting application source into a container image and assembling a new image. The new image incorporates the base image (the builder) and built source and is ready to use with the docker run command. S2I supports incremental builds, which re-use previously downloaded dependencies, previously built artifacts, etc.
+		
+		* pros:
+			* separating code and image development - probably advantage in bigger projects where development and devops is separated
+		* cons:
+			* not copying .git to image, using default paths (/opt/app/src), maybe others which can probably require our scripts rework
+			
+	* normal image build (eg. `docker build ...`):
+		* pros:
+			* already in use in our environment
+			* no need to implement additional tool
+			* clean approach
+
+| ---------------------------------- | ----------- | ------------ |
+| image name                         | build time  | size         |
+| ---------------------------------- | ----------- | ------------ |
+| ps_normal_github_base              | 4m42.531s   | 797MB        |
+| ps_normal_github_final             | 0m12.925s   | 797MB        |
+| ---------------------------------- | ----------- | ------------ |
+| ps_normal_rpm_base                 | 5m8.026s    | 804MB        |
+| ps_normal_rpm_final                | 0m15.766s   | 804MB        |
+| ---------------------------------- | ----------- | ------------ |
+| ps_s2i_github_base                 | 4m38.620s   | 1.12GB       |
+| ps_s2i_github_final                | 0m10.715s   | 1.12GB       |
+| ---------------------------------- | ----------- | ------------ |
+
+### High level build flow strategies
+
+OpenShift vs. Public service
+	* OpenShift:
+		* pros:
+			* flexible
+			* no waiting in queue
+		* cons:
+			* additional maintenance
+			* additional resources = additional costs/failures because of resource limit
+	
+	* Public service:
+		* pros:
+			* our current approach - no radical changes required
+			* simple and straightforward
+			* free
+			
+		* cons:
+			* dependency on external service
+			* long queue time in case of service issue/high load
+
+#### OpenShift
+	* Source-to-Image -
+		 Source-to-Image (`sourceStrategy` in OC configuration) is using s2i tool for image deployment
+	
+	Links:
+	* [s2i details](https://docs.openshift.com/container-platform/3.11/architecture/core_concepts/builds_and_image_streams.html#source-build)
+	* [s2i strategy options](https://docs.openshift.com/container-platform/3.11/dev_guide/builds/build_strategies.html#source-to-image-strategy-options)
+	* [s2i tool blog](https://www.openshift.com/blog/create-s2i-builder-image)
+
+	* Image Stream - current configuration
+	* Custom Build - need more details
+
+## Conclusion
+
+	* Source-to-image (S2I) - no performance improvements, bigger image
+	* RPM vs github:
```

Comment:
This section is relevant for `[python]-packit` and `python-ogr`, right?

Could/should we treat these two packages separately?

When installing from source with `pip` we should pay attention to how we pull in dependencies from PyPI, as those packages are not signed. Without enabling extra checks, this can become a fairly easy attack vector.

Reply:
is here option to install 3rd party via rpm (where applicable - some are using pip install as workaround) and install our projects via pip with  --no-deps?

Reply:
As discussed today:
- We have some dependencies both on PyPI and in Fedora and some Fedora only. => It would be nice to use only one source, the Fedora one.
  - Also for other reasons like signing as Hunor said or quality (packages in Fedora has fewer problems with installations than the PyPI ones since it's prepared for our environment and can use Fedora dependencies).
  - Another positive point is matching with packit-cli. We already had some problems with rebase-helper when Fedora and PyPI version differed.
- We should avoid adding a new place where we need to manage dependencies.
- We can:
  - Make a local build of Packit (via packit-cli or manually).
  - Try to install only dependencies using something like [this](https://github.com/packit-service/packit/blob/master/files/tasks/build-rpm-deps.yaml) and use `pip` with `--no-deps`.
    - note: Are `builddeps` enough?

Question for others: which version of ogr we want to use? Is `master` (the current state) ok?

Reply:
>     * Are `builddeps` enough?

No, they're not. But I have one heretic idea. What about adding those missing
```
Requires:       fedpkg
Requires:       rpmdevtools
Requires:       python3-bodhi-client
```
as
```
# These are technically not needed for the build,
# but we (packit team) need to have them here to be able to install all dependencies with
# dnf -y builddep packit.spec
BuildRequires:       fedpkg
BuildRequires:       rpmdevtools
BuildRequires:       python3-bodhi-client
```
as well into the [spec](https://github.com/packit-service/packit/blob/master/packit.spec) as well.
I don't think this (having more build requires than needed) is against [packaging guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines/#buildrequires).

Reply:
@jpopelka There is no `requires` version of that `dnf` command, is it?

I am not against your change otherwise. (And since we want to run real tests in the `test` step, we need to have those dependencies in all cases.)

Reply:
It might create a lot of confusion in the future if we start mixing build requirements with run-time requirements. These to categories overlapping is more accidental than intentional, so relying on it could cause unnecessary fragility.

Another approach we could take:

With the help of `rpmspec` one can get the run-time requirements from the spec file with

```
rpmspec -q --requires packit.spec | grep -v 'packit'
```

The grep is to get rid of `python3-packit` as a dependency, as that one is provided by us anyways.

This way we could get the run-time requirements that are specified only in the spec file.

We would still miss run-time requirements from `setup.cfg`. These could be obtained by running `python setup.py egg-info`, which would give us `packitos.egg-infor/requires.txt`. From this file we could install each run-time dependency as `dnf install python3dist(<dep>)`, of course after dropping the `[testing]` section. This mechanism would be similar to what Fedora does when [auto-generates Python dependencies](https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#_automatically_generated_dependencies).

Reply:
Looks like local build will be the easiest. (We can run `rpmbuild` directly if we don't want to install packit.)

It will avoid duplicating dependencies definition, parsing and other magic which can be confusing and can have errors.